Lab/demo/Lab/Plugins/IO/OM_io_plugin.cpp

```diff
@@ -0,0 +1,175 @@
+#include "SMesh_type.h"
+#include "Scene_surface_mesh_item.h"
+#include "Scene_polygon_soup_item.h"
+#include "Kernel_type.h"
+#include "Scene.h"
+
+#include <CGAL/Three/CGAL_Lab_io_plugin_interface.h>
+#include <CGAL/Three/Three.h>
+#include "Scene_polyhedron_selection_item.h"
+
+#include <CGAL/IO/OM.h>
+#include <CGAL/boost/graph/io.h>
+
+#include <QColor>
+#include <QString>
+#include <QStringList>
+#include <QMainWindow>
+#include <QInputDialog>
+
+#include <boost/property_map/function_property_map.hpp>
+
+#include <cstdint>
+#include <fstream>
+#include <iostream>
+#include <limits>
+#include <vector>
+
+using namespace CGAL::Three;
+class CGAL_Lab_om_plugin :
+  public QObject,
+  public CGAL_Lab_io_plugin_interface
+{
+  Q_OBJECT
+  Q_INTERFACES(CGAL::Three::CGAL_Lab_io_plugin_interface)
+  Q_PLUGIN_METADATA(IID "com.geometryfactory.CGALLab.IOPluginInterface/1.90" FILE "om_io_plugin.json")
+
+public:
+  QString nameFilters() const;
+  QString name() const { return "om_plugin"; }
+  bool canLoad(QFileInfo fileinfo) const;
+  QList<Scene_item*> load(QFileInfo fileinfo, bool& ok, bool add_to_scene=true);
+
+  bool canSave(const CGAL::Three::Scene_item*);
+  bool save(QFileInfo fileinfo,QList<CGAL::Three::Scene_item*>&);
+};
+
+QString CGAL_Lab_om_plugin::nameFilters() const {
+  return "om files (*.om)";
+}
+
+bool CGAL_Lab_om_plugin::canLoad(QFileInfo) const {
+  return true;
+}
+
+
+
+QList<Scene_item*>
+CGAL_Lab_om_plugin::
+load(QFileInfo fileinfo, bool& ok, bool add_to_scene){
+
+  // Open file
+  std::ifstream in(fileinfo.filePath().toUtf8(), std::ios::in | std::ios::binary);
+  if(!in) {
+    std::cerr << "Error! Cannot open file " << (const char*)fileinfo.filePath().toUtf8() << std::endl;
+    ok = false;
+    return QList<Scene_item*>();
+  }
+  in.close();
+  if(fileinfo.size() == 0)
+  {
+    CGAL::Three::Three::warning( tr("The file you are trying to load is empty."));
+    Scene_surface_mesh_item* item = new Scene_surface_mesh_item();
+    item->setName(fileinfo.completeBaseName());
+    ok = true;
+    if(add_to_scene)
+      CGAL::Three::Three::scene()->addItem(item);
+    return QList<Scene_item*>()<<item;
+  }
+
+  try
+  {
+    typedef boost::graph_traits<SMesh>::vertex_descriptor vertex_descriptor;
+    typedef boost::graph_traits<SMesh>::edge_descriptor edge_descriptor;
+    std::map<vertex_descriptor,bool> sm_vfeature_map;
+    auto sm_vfeature_pmap = boost::make_assoc_property_map(sm_vfeature_map);
+
+    std::map<edge_descriptor,bool> sm_efeature_map;
+    auto sm_efeature_pmap = boost::make_assoc_property_map(sm_efeature_map);
+
+    // Try building a surface_mesh
+    SMesh* SM = new SMesh();
```

Comment:
```suggestion
    SMesh* sm = new SMesh();
```
the name should be lower case

Reply:
I started from the STL plugin.